### PR TITLE
INSTUI-3032 refactor(ui-axe-check): better TypeScript types

### PR DESCRIPTION
**Pull request by matyasf**

### CODE REVIEW

File: packages/ui-axe-check/src/runAxeCheck.ts [hunk 1]
```diff
@@ -21,30 +21,39 @@
...
+  element: string,
+  options: AxeCheckOptions = {}
+) {
+  let result: Error | true = true
+  const context: axeCore.ContextObject = {
     include: [element],
-    // @ts-expect-error ts-migrate(2339) FIXME: Property 'exclude' does not exist on type '{}'.
     exclude: options.exclude
   }
   const config = {
```
Comment (ke1k0) [inline packages/ui-axe-check/src/runAxeCheck.ts:45]: I'd rather do this than a cast:
```suggestion
  const config: axeCore.RunOptions = {
```

File: packages/ui-axe-check/src/runAxeCheck.ts [hunk 2]
```diff
@@ -21,30 +21,39 @@
...
+  exclude?: string | string[]
+  /**
+   * Violations to ignore/filter out. 'color-contrast' is automatically excluded
+   * because we test color contrast in theme tests.
+   */
+  ignores?: string[]
+}
+
+export default async function runAxe(
+  element: string,
```
Comment (ke1k0) [inline packages/ui-axe-check/src/runAxeCheck.ts:37]: From the build error it seems that we pass HTMLElements to the `runAxeCheck(element, options)` in the ui-test-queries `accessible` util.

File: packages/ui-axe-check/src/runAxeCheck.ts [hunk 3]
```diff
@@ -79,20 +87,19 @@ export default async function runAxe(element, options = {}) {
...
+        violation.nodes
+          .map(function (node) {
+            return node.target.toString()
+          })
+          .join('\n'),
+        violation.description,
+        `${violation.helpUrl}\n`
+      ].join('\n')
+    })
+    .join('\n')
```
Comment (ke1k0) [inline packages/ui-axe-check/src/runAxeCheck.ts:104]: Is this a bug fix?

Comment (matyasf) [reply packages/ui-axe-check/src/runAxeCheck.ts:104]: yep, it had the wrong type

File: packages/ui-axe-check/src/runAxeCheck.ts (hunk 1 above)
Comment (matyasf) [reply packages/ui-axe-check/src/runAxeCheck.ts:45]: fixed

File: packages/ui-axe-check/src/runAxeCheck.ts (hunk 2 above)
Comment (matyasf) [reply packages/ui-axe-check/src/runAxeCheck.ts:37]: its a bug in axe-check's typing I've opened a bug and did a cast as a workaround

File: packages/ui-axe-check/src/runAxeCheck.ts [hunk 4]
```diff
@@ -21,30 +21,39 @@
...
  */
+
 import axeCore from 'axe-core'
 
-// @ts-expect-error ts-migrate(7006) FIXME: Parameter 'element' implicitly has an 'any' type.
-export default async function runAxe(element, options = {}) {
-  let result = true
-  const context = {
+type AxeCheckOptions = {
+  exclude?: string | string[]
```
[the remark this answers is not in the document]
Comment (matyasf) [reply packages/ui-axe-check/src/runAxeCheck.ts:28]: fixed it

File: packages/ui-axe-check/src/runAxeCheck.ts (hunk 2 above)
Comment (ke1k0) [reply packages/ui-axe-check/src/runAxeCheck.ts]: oh cool, thanks!

**Review by ke1k0 — APPROVED** [review]

**Pull request merged**